### PR TITLE
[INFRA-1176] Using repo.azure.jenkins.io as a mirror for CI builds

### PR DESCRIPTION
**Pull request by jglick**
# Description

[INFRA-1176](https://issues.jenkins-ci.org/browse/INFRA-1176)

Complements #2889. (I do not think the Maven mirror will automatically redirect those downloads; would need to define Maven properties for each special repository location so `Jenkinsfile` could switch those to a mirror. Lower priority than getting regular Maven artifacts cached.)

### Changelog entries

N/A

### Desired reviewers

@reviewbybees & @olblak

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -14,9 +14,7 @@ def runTests = true
 def failFast = false
 
 // Only keep the 10 most recent builds.
-properties([[$class: 'jenkins.model.BuildDiscarderProperty', strategy: [$class: 'LogRotator',
-                                                                        numToKeepStr: '50',
-                                                                        artifactNumToKeepStr: '20']]])
+properties([buildDiscarder(logRotator(numToKeepStr: '50', artifactNumToKeepStr: '20'))])
```
Comment (jglick) [inline Jenkinsfile]: Unrelated, just doing a little cleanup while I am here.

**Comment by jglick** [comment]
Not working, will follow up in JIRA.

File: Jenkinsfile [hunk 2]
```diff
@@ -14,9 +14,7 @@ def runTests = true
 def failFast = false
 
 // Only keep the 10 most recent builds.
```
Comment (olblak) [inline Jenkinsfile]: Is this comment still relevant?

**Review by oleg-nenashev — APPROVED** [review]
:bee:

File: Jenkinsfile (hunk 2 above)
Comment (jglick) [inline Jenkinsfile]: No I guess that is out of date.

**Comment by jglick** [comment]
@reviewbybees done

**Comment by daniel-beck** [comment]
@jglick The side effect of this merge is that untrusted PRs branched off from master from before this was merged will now fail, like #2904 or #2903. Can this be made conditional on the existence of the file?

**Comment by jglick** [comment]
Bug in Jenkins. I filed [JENKINS-44562](https://issues.jenkins-ci.org/browse/JENKINS-44562) for analysis. Unfortunately since the authors of those two PRs already followed your advice to merge with `master` (or rebase), it is now impossible to diagnose in those cases. I did check

```bash
$ curl -u … https://api.github.com/repos/jenkinsci/jenkins/pulls/2891
```

and it returned

```json
  "base": {
    "label": "jenkinsci:master",
    "ref": "master",
    "sha": "a8994d4fa55eb4f5ba1b5a60837876c8107eeb95",
```

which is to say, an old `.base.sha` as I would expect.

**Pull request merged**